pkg/cli/admin/release/extract_tools.go

```diff
@@ -297,16 +297,115 @@ func (o *ExtractOptions) extractCommand(command string) error {
 	infoOptions := NewInfoOptions(o.IOStreams)
 	infoOptions.SecurityOptions = o.SecurityOptions
 	infoOptions.FileDir = o.FileDir
-	release, err := infoOptions.LoadReleaseInfo(o.From, false)
+
+	targetRelease := targetReleaseInfo{}
+	typedReleaseRef, err := infoOptions.LoadReleaseInfo(o.From, false)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not load release %s info: %v", o.From, err)
+	}
+	targetRelease, err = o.setReleaseLookup(typedReleaseRef, targets, currentOS, willArchive)
+	if err != nil {
+		return fmt.Errorf("failed lookup of release %s: %v", o.From, err)
+	}
+	if targetRelease.willArchive {
```

Comment:
I'd suggest the change which re-shuffles this code from one place to be re-usable to be in a separate commit, it'll be easier to grasp and understand.

Reply:
updated, split this change into its own commit